tests/test_manual_urdf.py

```diff
@@ -0,0 +1,111 @@
+    mat1 = ET.SubElement(visual1, "material", {"name": "Part_1_color"})
+    ET.SubElement(mat1, "color", {"rgba": "0.8 0.4 0.1 1.0"})
+
+    # 4) Revolute_2 joint
+    joint_rev2 = ET.SubElement(root, "joint", {"name": "Revolute_2", "type": "revolute"})
+    ET.SubElement(joint_rev2, "origin", {"xyz": "-0.015 0.0025 -0.0012", "rpy": "1.5708 0 1.5708"})
+    ET.SubElement(joint_rev2, "parent", {"link": "Part_1"})
+    ET.SubElement(joint_rev2, "child", {"link": "Part_1_2"})
+    ET.SubElement(joint_rev2, "axis", {"xyz": "0 0 1"})
+    ET.SubElement(joint_rev2, "limit", {"effort": "80", "velocity": "5", "lower": "-1.5707963", "upper": "1.5707963"})
+
+    # 5) Part_1_2 link (visual only)
+    link_part1_2 = ET.SubElement(root, "link", {"name": "Part_1_2"})
+    visual2 = ET.SubElement(link_part1_2, "visual")
+    ET.SubElement(visual2, "origin", {"xyz": "0 0 0", "rpy": "0 0 0"})
+    geom2 = ET.SubElement(visual2, "geometry")
+    ET.SubElement(geom2, "mesh", {"filename": "meshes/Part_1_2.stl"})
+    mat2 = ET.SubElement(visual2, "material", {"name": "Part_1_2_color"})
+    ET.SubElement(mat2, "color", {"rgba": "0.7 0.6 0.3 1.0"})
+
+    # 6) Revolute_3 joint
+    joint_rev3 = ET.SubElement(root, "joint", {"name": "Revolute_3", "type": "revolute"})
+    ET.SubElement(joint_rev3, "origin", {"xyz": "0 0.0025 -0.014", "rpy": "3.1415927 0 0"})
+    ET.SubElement(joint_rev3, "parent", {"link": "Part_1"})
+    ET.SubElement(joint_rev3, "child", {"link": "Part_1_3"})
+    ET.SubElement(joint_rev3, "axis", {"xyz": "0 0 -1"})
+    ET.SubElement(joint_rev3, "limit", {"effort": "80", "velocity": "5", "lower": "-1.5707963", "upper": "1.5707963"})
+
+    # 7) Part_1_3 link (visual only)
+    link_part1_3 = ET.SubElement(root, "link", {"name": "Part_1_3"})
+    visual3 = ET.SubElement(link_part1_3, "visual")
+    ET.SubElement(visual3, "origin", {"xyz": "0 0 0", "rpy": "0 0 0"})
+    geom3 = ET.SubElement(visual3, "geometry")
+    ET.SubElement(geom3, "mesh", {"filename": "meshes/Part_1_3.stl"})
+    mat3 = ET.SubElement(visual3, "material", {"name": "Part_1_3_color"})
+    ET.SubElement(mat3, "color", {"rgba": "0.5 0.5 0.8 1.0"})
+
+    # Pretty-print (Python >=3.9). If older, comment out or remove.
+    ET.indent(root, space="  ", level=0)
+    tree = ET.ElementTree(root)
+    tree.write(urdf_file, encoding="unicode", xml_declaration=True)
+
+
+def _test_run_visualization_generated_simplified_adhoc() -> None:
+    """Run ad-hoc test with interactive visualization.
+
+    Directly call visualize_krec in interactive mode (no output path).
+    Launches the GUI so you can confirm the model visually.
+    """
+    # Create a local URDF file
+    temp_urdf = Path(URDF_BASE_PATH / "manual_example.urdf")
+    create_simplified_urdf(temp_urdf)
+
+    visualize_krec(krec_path=None, urdf_path=temp_urdf)
+
+
+if __name__ == "__main__":
+    # Run an ad-hoc test that shows the GUI (rather than the auto test).
+    _test_run_visualization_generated_simplified_adhoc()
```

Comment:
@budzianowski can you take a look at this test?

I understand that you want to confirm that the final rotations are correct/incorrect. That info is already there in the logs and I have confirmed many times that both python and rust are logging the same transforms. But anyways I will still add an explicit check